### PR TITLE
Fixed issue #18669: XSS simple admin via Google key

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

**Comment by Shnoulle** [comment]
OK for new one, 
But must fix in Vanilla theme too : filter value before using it.

**Comment by gabrieljenik** [comment]
> But must fix in Vanilla theme too : filter value before using it.

Filter what? if it is already filtered.

**Review by Shnoulle — APPROVED** [review]

File: application/models/Survey.php [hunk 1]
```diff
@@ -552,6 +552,11 @@ public function rules()
             array('startdate', 'date','format' => ['yyyy-M-d H:m:s.???','yyyy-M-d H:m:s','yyyy-M-d H:m'],'allowEmpty' => true),
             array('datecreated', 'date','format' => ['yyyy-M-d H:m:s.???','yyyy-M-d H:m:s','yyyy-M-d H:m'],'allowEmpty' => true),
             array('expires', 'checkExpireAfterStart'),
+            // The Google Analytics Tracking ID is inserted in a JS script. If the following rule is changed, make sure
+            // that it doesn't render it vulnerable to XSS attacks.
+            array('googleanalyticsapikey', 'match', 'pattern' => '/^[a-zA-Z\-\d]*$/',
+                'message' => gT('Google Analytics Tracking ID may only contain alphanumeric characters and hyphens.'),
+            ),
```
Comment (c-schmitz) [inline application/models/Survey.php:559]: Is 'allowEmpty' missing here?

**Review by c-schmitz — CHANGES_REQUESTED** [review]

File: application/models/Survey.php (hunk 1 above)
Comment (gabrieljenik) [reply application/models/Survey.php:559]: Do we need one if the pattern allows empty (there is a star near the end of the pattern)?

**Comment by Shnoulle** [comment]
Needed for 3.X and 5.X right ?

**Comment by Shnoulle** [comment]
Confirm : still XSS issue in 3.X and 5.X

**Comment by tiborpacalat** [comment]
If it is still an issue then we need to port this fix to 3 and 5 as well.

**Comment by Shnoulle** [comment]
> If it is still an issue then we need to port this fix to 3 and 5 as well.

Yes, i put a comment on mantis

**Pull request merged**